contrib/opentelemetry/tracing_interceptor.go

```diff
@@ -140,46 +140,54 @@ func (t *tracer) Options() interceptor.TracerOptions {
 }
 
 func (t *tracer) UnmarshalSpan(m map[string]string) (interceptor.TracerSpanRef, error) {
-	ctx := trace.SpanContextFromContext(t.options.TextMapPropagator.Extract(context.Background(), textMapCarrier(m)))
-	if !ctx.IsValid() {
+	ctx := t.options.TextMapPropagator.Extract(context.Background(), textMapCarrier(m))
+	spanCtx := trace.SpanContextFromContext(ctx)
+	if !spanCtx.IsValid() {
 		return nil, fmt.Errorf("failed extracting OpenTelemetry span from map")
 	}
-	return &tracerSpanRef{SpanContext: ctx}, nil
+	return &tracerSpanRef{SpanContext: spanCtx, Baggage: baggage.FromContext(ctx)}, nil
```

Comment:
Sorry to sound tedious, but can we add an option to at least opt-out of baggage (i.e. `TracerOptions.DisableBaggage`? That way if someone comes and says "hey, why did you start sending my baggage across and change behavior?" we can at least tell them they can disable it?

Reply:
Absolutely! I'll make that change today